Lib/contextlib.py

```diff
@@ -54,8 +54,9 @@ def inner(*args, **kwds):
         return inner
 
 
-class _GeneratorContextManager(ContextDecorator, AbstractContextManager):
-    """Helper for @contextmanager decorator."""
+class _GeneratorContextManagerBase(ContextDecorator):
```

Comment:
You're right, my implementation didn't actually work as a decorator. (Although it's because ContextDecorator uses `with` instead of `async with`, not because of anything to do with `__call__`.)

We could add a separate `AsyncContextDecorator` implementation, but I don't currently have a good use for that.